src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -15,12 +16,23 @@ void DmHina_Destroy(Actor* thisx, PlayState* play);
 void DmHina_Update(Actor* thisx, PlayState* play);
 void DmHina_Draw(Actor* thisx, PlayState* play);
 
-void func_80A1F470(DmHina* this, PlayState* play);
-void func_80A1F56C(DmHina* this, PlayState* play);
-void func_80A1F5AC(DmHina* this, PlayState* play);
-void func_80A1F63C(DmHina* this, PlayState* play);
+void DmHina_Idle(DmHina* this, PlayState* play);
+void DmHina_AwaitMessageBoxClosing(DmHina* this, PlayState* play);
+void DmHina_SetupSubCamera(DmHina* this, PlayState* play);
+void DmHina_MoveSubCamera(DmHina* this, PlayState* play);
+
+typedef enum {
+    /* 0 */ PARAMS_REMAINS_ODOLWA,
+    /* 1 */ PARAMS_REMAINS_GOHT,
+    /* 2 */ PARAMS_REMAINS_GYORG,
+    /* 3 */ PARAMS_REMAINS_TWINMOLD
+} DmHinaType;
```

Comment:
These need to be in the header, they should have an actor-specific prefix, they should use TYPE instead of PARAMS, and `func_808B849C` returns this thing + 1 from the looks of it.

Reply:
There's also a bug that needs marking in DoorWarp1 which was why this actor was interesting in the first place, I defer to @fig02 for the details since he did the research.